src/coreclr/nativeaot/BuildIntegration/Microsoft.NETCore.Native.Unix.props

```diff
@@ -95,6 +95,8 @@ The .NET Foundation licenses this file to you under the MIT license.
       <LinkerArg Include="-Wl,-z,relro" Condition="'$(TargetOS)' != 'OSX'" />
       <!-- binskim warning BA3011 The BIND_NOW flag is missing -->
       <LinkerArg Include="-Wl,-z,now" Condition="'$(TargetOS)' != 'OSX'" />
+      <!-- binskim warning BA3006: The non-executable stack is not enabled -->
+      <LinkerArg Include="-Wl,-z,noexecstack" Condition="'$(TargetOS)' != 'OSX'" />
```

Comment:
We do not pass this option when building libcoreclr.so, but still get noexecstack. 

I think we may be missing NoExecStack bit in the .o file: https://github.com/llvm/llvm-project/blob/cb395f66ac3ce60427ca2b99580e716ac6dd551a/llvm/lib/MC/MCELFStreamer.cpp#L97

Overriding NoExecStack bit using a linker options gets the job done - but it would be nice to emit it correctly in the .o file too.

Reply:
We pass them in global scope only for assembly code: https://github.com/dotnet/runtime/blob/b7e0a85ab6111b77ad19a1c2db0dedcedded4940/eng/native/configurecompiler.cmake#L187-L195
In mono, we pass them only on Android (for all languages C/assembly) https://github.com/dotnet/runtime/blob/0e800a6139f1a8410c8ac6d52f5e619c446fa23e/src/mono/CMakeLists.txt#L219

Reply:
Oh, I see. I now [read up](https://wiki.gentoo.org/wiki/Hardened/GNU_stack_quickstart) about it.

So we need to place a GNU-stack note in the object file produced by the NativeAOT compiler and it should ideally Just Work™ after that with no linker switches necessary. Yes, looks like the fix is going to be in the dotnet/llvm-project again.

I don't see a need to ever not set it for what NativeAOT produces, so I won't be making it an option for the compiler and make it unconditionally emitted.